### DISCUSSION TITLE
Making Contributions

### DISCUSSION THREAD

**Discussion by IAmRiteshKoushik** [pocketbase/pocketbase#2815] [Q&A] [answered]
Hey @ganigeorgiev . I have noticed that there are almost 40+ feature requests on pocketbase. I would like to work on some of them as part of beginning to contribute to Open Source. Could you and the community perhaps streamline it down a bit and help me out a little here with ? Listing down beginner-friendly issues would be really helpful. Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
Just because there is an open feature request it doesn't necessary mean that it needs to be implemented. If it was something easy and clear it would have been done already.

Contributions for bug fixes, code optimizations and docs improvements are more than welcomed.

But please refrain working on PRs for **new features** without previously discussing the implementation details because I don't have a clear idea how the existing feature requests will be integrated, how they will play with each other and whether they need to be part of PocketBase in the first place. Every feature comes with a cost and at this stage I don't want to be burdened with the obligation that is implied when dealing with spare/random contributions.

PocketBase has a [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1) and I prefer to work on tasks in specific order and random PRs for new features doesn't help as they require focus shift, tedious back-and-forth communication and as mentioned previously, often I'm not sure of the correct approach and end up experimenting with different versions before choosing the one that I think would fit best.

To summarize: I don't need help with contributions for new features. If you find something that can be optimized or fixed, feel free to create a PR, but if you just want to contribute to an open source project it may be better to look somewhere else where your effort and time will be more needed.
